packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -682,11 +700,24 @@ export class GarbageCollector implements IGarbageCollector {
      * We current write the entire GC state in a single blob. This can be modified later to write multiple
      * blobs. All the blob keys should start with `gcBlobPrefix`.
      */
-    public summarize(): ISummaryTreeWithStats | undefined {
+    public summarize(fullTree: boolean, trackState: boolean): ISummaryTreeWithStats | undefined {
         if (!this.shouldRunGC || this.previousGCDataFromLastRun === undefined) {
             return;
         }
 
+        const builder = new SummaryTreeBuilder();
+
+        if (
+            this.gcDataHasNotChangedFromLastRun &&
```

Comment:
The second point here is very crucial. In case of a failed summary, we have to make sure that the next GC run compares with the data from the last successful summary.
Basically, the state that tracks this should only be updated if the summary was successful. Take a look at `latestSummaryStateRefreshed` which is called when a summary ack is received and has properties telling whether the last summary state should be updated.